pysd/py_backend/vensim/vensim2py.py

```diff
@@ -350,6 +365,50 @@ def get_external_data(func_str, args_str, root_path):
 
     return f(*args, root=root_path).subscript
 
+def get_subscript_numeric_range(subs_start, subs_end):
+    """
+    With the first and the last subscript values of a subscript
+    numeric range, gets the common prefix of both and the 
+    starting and ending number of the numeric range
+
+    Parameters
+    ----------
+    subs_start: str
+        Represents the first subscript value in the numeric range
+    subs_end: str
+        Represents the last subscript value in the numeric range
+    
+    Returns
+    -------
+    prefix: str
+        Common prefix of both subscripts
+    num_start: int
+        Sequence start number 
+    num_end: int
+        Sequence end number
+    
+    Examples
+    --------
+    >>> get_subscript_number_range('Layer1', 'Layer5')
+    ('Layer', 1, 5)
+    >>> get_subscript_number_range('sub15', 'sub30')
+    ('sub', 15, 30)
+    """
+    if(subs_start == subs_end): raise ValueError('Only different subscripts are valid in a numeric range')
+
+    subs_start = re.findall('\d+|\D+', subs_start)
+    subs_end = re.findall('\d+|\D+', subs_end)
+    prefix_start = ''.join(subs_start[:-1])
+    prefix_end = ''.join(subs_start[:-1])
+    num_start = int(subs_start[-1])
+    num_end = int(subs_end[-1])
+
+    if(not(prefix_start) or not(prefix_end)): raise ValueError('A numeric range must contain at least one letter')
+    if(num_start>num_end): raise ValueError('The number of the first subscript value must be lower than the second subscript value in a subscript numeric range\n')
+    if(prefix_start != prefix_end or subs_start[0].isdigit() or subs_end[0].isdigit()): raise ValueError('Only matching names ending in numbers are valid')
```

Comment:
This seems nice :)
I have had some headaches about how to give human-readable errors when something breaks inside the parser.
As when an error is raised parsimonious usually rises own errors, which sometimes are difficult to follow and remove the information of what failed, this is more common in parse_general_expression parser.
Have you tried raising these errors when parsing a file?

I added some more informative messages in the parser in PR #249 (see lines 347-359 and 1121-1134 from vensim2py). We may need to see how to incorporate all error raises together so they are easy to identify.

Reply:
Yes, I have tried raising these errors when parsing a file.
I am going to add more information in these error messages so that the user can identify in which numeric range name the error is found in more easily.